pkg/descheduler/strategy_migration.go

```diff
@@ -263,9 +238,24 @@ var pluginsMap = map[string]func(ctx context.Context, nodes []*v1.Node, params *
 			klog.V(1).ErrorS(err, "unable to initialize a plugin", "pluginName", nodeutilization.LowNodeUtilizationPluginName)
 			return
 		}
-		status := pg.(framework.BalancePlugin).Balance(ctx, nodes)
+		RunPlugin(pg, pg.(framework.BalancePlugin).Balance, tracing.BalanceOperation, nodeutilization.LowNodeUtilizationPluginName)(ctx, nodes)
+	},
+}
+
+// RunFunc represents a framework.Plugin function.
+// - (framework.BalancePlugin).Balance
+// - (framework.DeschedulePlugin).Deschedule
+type RunFunc func(ctx context.Context, nodes []*v1.Node) *framework.Status
+
+// RunPlugin runs a plugin.
+func RunPlugin(plugin framework.Plugin, runFn RunFunc, operationName, pluginName string) func(ctx context.Context, nodes []*v1.Node) {
```

Comment:
rather than wrapping the plugin's function in a span, why not start and close the span within the plugin's `Deschedule` function?

I would prefer if traces are implemented by the plugins themselves rather than the framework. The framework can provide the mechanisms (like registering a global traceprovider) but I would leave the actual spans up to plugin developers to implement

Reply:
My initial attempt was to put Span login within each plugin individually. And then I thought it could hard to maintain if we manage tracing stuff separately:
* it would be easy to forget tracing logic in the function
* it would bring code duplication (1 line tracing, 1 line defer _times_ plugin count)
* it would be easy to forget to add `span.RecordError(status.Err)` function in all error cases
* should plugins really know the existence of tracing? (why plugin developers should duplicate this logic?)

These are just my concern, I'm OK to go with your way anyway. Wanted to learn more about your concern. 🤞

Reply:
> should plugins really know the existence of tracing? (why plugin developers should duplicate this logic?)

One benefit of a framework model is detangling the plugin logic from the surrounding framework internals, so in that sense the answer is yes.

On that end, it makes sense for us to start a parent trace in our internals (somewhere around [here](https://github.com/kubernetes-sigs/descheduler/blob/v0.25.1/pkg/descheduler/descheduler.go#L355)), and plugins can use the context to add whatever additional spans they want. But the responsibilities of good design you mention in your other bullet points are kept to the individual plugin developers

Reply:
> On that end, it makes sense for us to start a parent trace in our internals (somewhere around [here](https://github.com/kubernetes-sigs/descheduler/blob/v0.25.1/pkg/descheduler/descheduler.go#L355)),

+1 from me for this. I think we can start from this for now and see how it goes. Enforcing the tracing related code can be done mostly during the code review of a new plugin getting added anyway.  The current code in the repo already lets us create the parent trace and pass the context down to the individual plugin/strategies so that they can trace the required bit of the code in that accordingly. 

This will also leave the plugin developer with freedom of being able to trace some of their internal helper/util functions as well if required.

Reply:
@harshanarayana yup, that's exactly what I'm thinking

Reply:
<img width="2048" alt="Screenshot 2022-10-04 at 4 18 24 PM" src="https://user-images.githubusercontent.com/1404867/193801065-50cf0605-b97e-485d-9834-801782de7a4b.png">